src/editor/adapters/in-memory-editor.ts

```diff
@@ -204,6 +208,87 @@ export class InMemoryEditor implements Editor {
   removeLine(line: number): void {
     this.codeMatrix.splice(line, 1);
   }
+
+  async getSelectionReferences(selection: Selection): Promise<CodeReference[]> {
```

Comment:
🍰 Ideally, I would like to write one or two contract tests to verify that this behavior is similar to what VS Code actually does. I don't write a lot of these tests since they are slower and more likely to fail, but just a handful is enough to cover the blind spots introduced with the In-Memory implementation.